modules/ingest-geoip/qa/full-cluster-restart/build.gradle

```diff
@@ -24,3 +25,10 @@ buildParams.bwcVersions.withWireCompatible { bwcVersion, baseName ->
     systemProperty("tests.old_cluster_version", bwcVersion)
   }
 }
+
+tasks.register("bcUpgradeTest", StandaloneRestIntegTestTask) {
```

Comment:
lets extract that task registration into a plugin script and avoid all this duplication of task registration logic. we do something simliar to bwc tests. 
see https://github.com/elastic/elasticsearch/blob/ae569def9c2cf95df3a642d431d8b8e63f6c9c19/build-tools-internal/src/main/groovy/elasticsearch.bwc-test.gradle as an example. you would just wrap this task registration lgoic into `build-tools-internal/src/main/groovy/elasticsearch.bc-update-test.gradle` and then apply the script as a plugin by ```
apply plugin: 'elasticsearch.bc-upgrade-test'
```